api/src/models/JobMetadata.ts

```diff
@@ -58,31 +60,61 @@ class JobMetadata {
     }
 
     get derivativeStatus(): Record<string, unknown> {
-        // TODO: populate with real data
+        let lockfileExists: boolean = fs.existsSync(this.derivativeLockfile);
         const status = {
             expected: 10,
             processed: 0,
             building: false,
+            lockfileExists
```

Comment:
lockfileExists should be the value of the building key; here it's just getting added as a free-floating element in the object.